swift/storage.py

```diff
@@ -426,3 +439,23 @@ def get_available_name(self, name, max_length=None):
         overwrite it.
         """
         return name
+
+
+class CachedStaticSwiftStorage(CachedFilesMixin, StaticSwiftStorage):
+    """
+    A static file system storage backend which also saves
+    hashed copies of the files it saves.
+    """
+    pass
+
+
+class ManifestStaticSwiftStorage(ManifestFilesMixin, StaticSwiftStorage):
+    """
+    A static file system storage backend which also saves
+    hashed copies of the files it saves.
+    """
+    def read_manifest(self):
+        try:
+            super(ManifestStaticSwiftStorage, self).read_manifest()
+        except swiftclient.ClientException:
+            return None
```

Comment:
Instead of silently swallowing the exception I would suggest to wrap it in `IOError` instead, like this (and maybe in the `StaticSwiftStorage._open` method instead of `ManifestStaticSwiftStorage.read_manifest `):

```python
except swiftclient.exceptions.ClientException as e:
    # django.contrib.staticfiles.storage.ManifestFilesMixin#read_manifest
    # When manifest file doesn't exist, the raised error must be an instance of IOError.
    raise IOError(e)
```

If you want to ignore the missing manifest error, it's possible by setting `ManifestStaticSwiftStorage.manifest_strict` property to `False`. So something like this will allow to make the ignoring of the error configurable:

```python
class ManifestStaticSwiftStorage(ManifestFilesMixin, StaticSwiftStorage):
    manifest_strict = setting('SWIFT_MANIFEST_STRICT', True)
```